platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/EmbeddedSolrFiles.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.solr.core.SolrConfig;
+import org.apache.solr.schema.IndexSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * Maintain embedded Solr configuration and schema files.
+ *
+ * <p><i>Note:</i> The corresponding {@link SolrConfig} and {@link IndexSchema} will be constructed
+ * the first time they are retrieved.
+ */
+public class EmbeddedSolrFiles {
+  private static final Logger LOGGER = LoggerFactory.getLogger(EmbeddedSolrFiles.class);
+
+  private final String coreName;
+  private final String configName;
+  private final File configFile;
+  private final String schemaName;
+  private final File schemaFile;
+  private volatile SolrConfig solrConfig = null;
+  private volatile IndexSchema schemaIndex = null;
+
+  /**
+   * Constructor.
+   *
+   * @param coreName name of the Solr core
+   * @param configXml name of the Solr configuration file
+   * @param schemaXmls file names of the Solr core schemas to attempt to load (will start with the
+   *     first and fallback to the others in order if unavailable)
+   * @param configProxy {@link ConfigurationFileProxy} instance to use
+   * @throws IllegalArgumentException if <code>coreName</code>, <code>configXml</code>, <code>
+   *     schemaXmls</code>, or <code>configProxy</code> is <code>null</code> or if unable to find
+   *     any files
+   */
+  public EmbeddedSolrFiles(
+      String coreName, String configXml, String[] schemaXmls, ConfigurationFileProxy configProxy) {
+    Validate.notNull(coreName, "invalid null Solr core name");
+    Validate.notNull(configXml, "invalid null Solr configuration file");
+    Validate.notNull(schemaXmls, "Invalid null Solr schema files");
+    Validate.notEmpty(schemaXmls, "missing Solr schema files");
+    Validate.noNullElements(schemaXmls, "invalid null Solr schema file");
+    Validate.notNull(configProxy, "invalid null Solr config proxy");
+    this.coreName = coreName;
+    this.configName = configXml;
+    this.configFile = EmbeddedSolrFactory.getConfigFile(configXml, configProxy, coreName);
+    Validate.notNull(configFile, "Unable to find Solr configuration file");
+    File solrSchemaFile = null;
+    String schemaXml = null;
+
+    for (final String s : schemaXmls) {
+      schemaXml = s;
+      solrSchemaFile = EmbeddedSolrFactory.getConfigFile(schemaXml, configProxy, coreName);
+      if (solrSchemaFile != null) {
+        break;
+      }
+    }
+    Validate.notNull(solrSchemaFile, "Unable to find Solr schema file.");
+    this.schemaFile = solrSchemaFile;
+    this.schemaName = schemaXml;
+  }
+
+  /**
+   * Gets the home directory where the configuration resides.
+   *
+   * @return the home directory where the configuration resides
+   */
+  public File getConfigHome() {
+    return configFile.getParentFile();
+  }
+
+  /**
+   * Creates or retrieves the schma index corresponding to this set of Solr files.
```

Comment:
✏️ Typo: _schema_

Reply:
⭕️ 
On the above comment for the stream (somehow I cannot add a response there), I cannot use streams as I needed to remember the schemaXml selected and not just the returned value from getResource(). The only way I could do it is to use an AomicReference object and set it from within the stream as a side effect which I wasn't too found of.

Reply:
👍